blog/content/post/2021-12-10-fobnail_2nd_phase.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Fobnail Token - developing communication method that meets the CHARRA requirements
+abstract: 'The Fobnail Token is an open-source hardware USB device that help to
+           determine the integrity of the system. The purpose of this blog post
+           is to present the development progress of this project. During the
+           last phase, we managed to implement the communication method that
+           will be used between verifier and attester.'
+cover: /covers/image-file.png
+author: tomasz.zyjewski
+layout: post
+published: true
+date: 2021-12-10
+archives: "2021"
+
+tags:
+  - fobnail
+  - firmware
+  - usb
+  - CHARRA
+categories:
+  - Firmware
+  - Security
+
+---
+
+# About the Fobnail Token project
+
+The Fobnail Token is a project that aims to provide a reference architecture for
+building offline integrity measurement servers on the USB device and clients
+running in Dynamically Launched Measured Environments (DLME). It allows the
+Fobnail owner to verify the trustworthiness of the running system before
+performing any sensitive operation. This project was founded by [NlNet
+Foundation](https://nlnet.nl/). More information about the project can be found
+in the [Fobnail documentation](https://fobnail.3mdeb.com/).
+
+# Communication in CHARRA
+
+[CHARRA](https://github.com/Fraunhofer-SIT/charra) is a "Challenge/Response
+Remote Attestation" interaction model of the IETF RATS Reference Interaction
+Models for Remote Attestation Procedures using TPM 2.0. In this project, the
+attester and verifier communicate with themselves using libcoap. In order to
+achieve that we need to implement Ethernet over USB on the Fobnail Token. We
+decided to use Rust so [nrf-hal](https://github.com/nrf-rs/nrf-hal) project
+provide us with a USB driver, and the conducted
+[research](https://fobnail.3mdeb.com/eth-over-usb-research/)
+allowed us to determine that EEM will be the most appropriate protocol
+implementing Ethernet over USB. Additionally, we use
+[smoltcp](https://github.com/smoltcp-rs/smoltcp) which is an interesting project
+that provides an implementation of TCP/IP stack.
+
+# Building applications for Fobnail
+
+We started our work on running the `hello-world` example using Rust nrf-hal. It
+turns out that the repository is missing an example for the nRF52840 which we
+use as a Fobnail prototype. The needed code can be found on
```

Comment:
@TomaszAIR This part is unclear. The seconds sentence mentions that it turns out that there is no example code for the platform we use, thats ok. But the next sentence says something about code, but it is unclear what is the needed code. We should mention here that we have ported the `blinky-demo` example to run on nRF52840, and then `The code can found in`